be/src/cache/datacache.cpp

```diff
@@ -173,9 +189,14 @@ StatusOr<CacheOptions> DataCache::_init_cache_options() {
             // we do not automatically rename it when the source and destination directories are on different disks.
             // In this case, users should manually remount the directories and restart them.
             std::string datacache_path = root_path.path + "/datacache";
-            std::string starlet_cache_path = root_path.path + "/starlet_cache/star_cache";
 #ifdef USE_STAROS
             if (config::datacache_unified_instance_enable) {
+                std::string starlet_cache_path;
+                if (idx < corresponding_starlet_dirs.size()) {
+                    starlet_cache_path = corresponding_starlet_dirs[idx++] + "/star_cache";
+                } else {
+                    starlet_cache_path = root_path.path + "/starlet_cache/star_cache";
+                }
```

Comment:
If the old starlet cache locates in `/data1/starlet_cache` and `/data3/starlet_cache`, this will cause the old cache data is moved to `/data1/${storage_root_path}/datacache` and `/data2/${storage_root_path}/datacache` ?